File: First and Second Smallests/First_and_Second_Smallests.py
class Solution:
    def minAnd2ndMin(self, arr):
        first = second = float('inf')
        for i in arr:
            if first > i:
                second = first
                first = i
            elif first < i and second > i:
                second = i
        if first == float('inf') or second == float('inf'): 
            return [-1]
        return [first,second];
        
  

File: First and Second Smallests/First_and_Second_Smallests.cpp
class Solution {
  public:
    vector<int> minAnd2ndMin(vector<int> &arr) {
        int first =INT_MAX,second = INT_MAX;
        int n = arr.size();
        for(int i = 0; i<n; i++) {
            if(first > arr[i]) {
                second = first;
                first = arr[i];
            }
            else if(first < arr[i] && second > arr[i]) {
                second = arr[i];
            }
        }
        if(first == INT_MAX || second == INT_MAX) return {-1};
        return {first,second};
    }
};